tests/providers/amazon/aws/hooks/test_rds.py

```diff
@@ -17,14 +17,301 @@
+    return db_cluster_snapshot["DBClusterSnapshotIdentifier"]
+
+
+@pytest.fixture
+def export_task_id(rds_hook: RdsHook, db_snapshot_arn: str) -> str:
+    response = rds_hook.conn.start_export_task(
+        ExportTaskIdentifier="testrdshook-export-task",
+        SourceArn=db_snapshot_arn,
+        S3BucketName="test",
+        IamRoleArn="test",
+        KmsKeyId="test",
+    )
+    return response["ExportTaskIdentifier"]
+
+
+@pytest.fixture
+def event_subscription_name(rds_hook: RdsHook, db_instance_id: str) -> str:
+    """Creates an mock RDS event subscription and returns its name"""
+    response = rds_hook.conn.create_event_subscription(
+        SubscriptionName="testrdshook-event-subscription",
+        SnsTopicArn="test",
+        SourceType="db-instance",
+        SourceIds=[db_instance_id],
+        Enabled=True,
+    )
+    return response["EventSubscription"]["CustSubscriptionId"]
+
+
 class TestRdsHook:
+    # For testing, set the delay between status checks to 0 so that we aren't sleeping during tests,
+    # and max_attempts to 1 so that we don't retry unless required.
+    waiter_args = {"check_interval": 0, "max_attempts": 1}
+
     def test_conn_attribute(self):
         hook = RdsHook(aws_conn_id="aws_default", region_name="us-east-1")
         assert hasattr(hook, "conn")
         assert hook.conn.__class__.__name__ == "RDS"
         conn = hook.conn
         assert conn is hook.conn  # Cached property
         assert conn is hook.get_conn()  # Same object as returned by `conn` property
+
+    def test_get_db_instance_state(self, rds_hook: RdsHook, db_instance_id: str):
+        response = rds_hook.conn.describe_db_instances(DBInstanceIdentifier=db_instance_id)
+        state_expected = response["DBInstances"][0]["DBInstanceStatus"]
+        state_actual = rds_hook.get_db_instance_state(db_instance_id)
+        assert state_actual == state_expected
+
+    def test_wait_for_db_instance_state_boto_waiters(self, rds_hook: RdsHook, db_instance_id: str):
+        """Checks that the DB instance waiter uses AWS boto waiters where possible"""
+        for state in ("available", "deleted"):
+            with patch.object(rds_hook.conn, "get_waiter") as mock:
+                rds_hook.wait_for_db_instance_state(db_instance_id, target_state=state, **self.waiter_args)
+                mock.assert_called_once_with(f"db_instance_{state}")
+                mock.return_value.wait.assert_called_once_with(
+                    DBInstanceIdentifier=db_instance_id,
+                    WaiterConfig={
+                        "Delay": self.waiter_args["check_interval"],
+                        "MaxAttempts": self.waiter_args["max_attempts"],
+                    },
+                )
+
+    def test_wait_for_db_instance_state_custom_waiter(self, rds_hook: RdsHook, db_instance_id: str):
+        """Checks that the DB instance waiter uses custom wait logic when AWS boto waiters aren't available"""
```

Comment:
Making sure custom waiter logic calls the generic waiter method, and the correct boto resource "describe" method